### ISSUE TITLE
loss becomes NAN

### ISSUE THREAD

**Issue by yxxxxxxxx** [sciencefans/Quality-Aware-Network#8]
Hello!I am a Postgraduate student who want to reproduce the PQAN_image Net on the LPW dataset . But my tripletloss and contrastive loss became Nan when training for several thousands iterations. I used your solver_PQAN_image.prototxt where the lr_rate=0.001 and train_LPW.m . Could you tell me what should I do to solve this problem, thank you very much!

**Comment by songguanglu** [comment]
The reason of NaN may be because 0 divided by 0 when the score is normalized(the scores of images belonging to the same person are 0).  There are some suggestions as follows:
1. If your loss keeps rising and finally becomes NaN I think you should disable memory optimization which may be not friendly to some structures.
2. Reducing the margin of triplet loss and using the pre-trained model.
3. If the hardware conditions support, increase batchsize.

**Comment by yxxxxxxxx** [comment]
Thanks for your suggestions! I will try those solutions!

**Comment by XinshaoWang** [comment]
Hi @songguanglu , it does not make sense to increase the batch size as I found that the length of some videos is smaller than 10. (I increased the batch size from 24 to 30, for some videos, the frame number is smaller than 10.)

**Comment by XinshaoWang** [comment]
@songguanglu , could you please represents how to reproduce the results on LPW?

**Comment by Duankaiwen** [comment]
Hello@XinshaoWang, @yxxxxxxxx, I meet the same problem, have you solved it?

**Comment by Duankaiwen** [comment]
I have solved the problem,set margin of tripletloss =1.0, this works for me

**Comment by yxxxxxxxx** [comment]
@songguanglu In your PRID2011 dataset,  I see you only select the id which has more than 27 frames, so it's only 178 ids not 200 ids in total which is different from the standard protocol, Can you tell me it's Ok to do that or I should choose 200 ids who appear in both cams? thank you very much!

**Comment by XinshaoWang** [comment]
Hi all, for PRID 2011 dataset, there is one mistake because of previous ECCV 2014 paper or TPAMI 2016 version "Person Re-Identification by Video Ranking". 
The number of persons with more than 21 frames is 181, rather than 178 as they said in the paper. 

That paper just used 178 persons rather than 181 persons or the whole dataset. 
From my observation, the number of persons with no less than 26 frames or no less than 27 frames is 178. 
That is why QAN also selects 178 persons who contain no less than 27 frames.  

By the way, according to my experiments and observations, QAN does not learn proper qualities, which I mean you can achieve similar performance dropping the quality learning component.

**Comment by yxxxxxxxx** [comment]
@XinshaoWang Hi,I learned a lot from your comments, "which I mean you can achieve similar performance dropping the quality learning component." you mean the QAN or PQAN?I tried the PQAN and achieve much better performance than the QAN and is it ok to do experiments on the 178 ids?

**Comment by XinshaoWang** [comment]
Hi @yxxxxxxxx , I opened another issue about the quality learning issue. 
If you read PQAN paper and its code carefully, you will find they are inconsistent, they used rigid regions instead of detected meaningful body regions, **which is really misleading**. 
In this case, PQAN is hard to address the misalignment problem. 
In my submission and coming code, we just use global information, which works much better.

**Comment by yxxxxxxxx** [comment]
@XinshaoWang Yeah I have noticed the PQAN split the image for 3:2:2 from the height. I think the author may want to put attention on the occlusion part and indeed，it may be hard to address the misalignment problem. I am looking forward to your submission and your new idea.

**Comment by XinshaoWang** [comment]
Hi @yxxxxxxxx , you are right, misalignment is a hard problem. 
I am just thinking what if someone else combines quality learning and body regions detection successfully.

**Comment by yxxxxxxxx** [comment]
@XinshaoWang I don't know if it is right but fundamentally speaking I think the quality learning is an attention mechanism from the perspective of batch dimension. So if we want to combines quality learning and body regions detection I think it is an spatial attention mechanism which I have seen a lot in the recent papers but it is a method I want to achieve to address the misalignment.

**Comment by XinshaoWang** [comment]
@yxxxxxxxx , yes your are right. 
QAN tries to apply image-level attention for set-based problems, there are some other papers using image-level attention like CaffeNet+SRM+TAM \cite{zhou2017see}, ASTPN \cite{xu2017jointly}. 

Besides, many papers apply region-level attention for image caption or image-based person ReID.

From my observations, the image-level attention for Video-based Person ReID does not work.
And I have done some work to make it work. But now I am not able to talk about it. 
Let's talk when my code is released.   Sorry.

Combining two kind of attention seems attractive.

**Comment by kikyou123** [comment]
@yxxxxxxxx, @Duankaiwwen    I meet the same problem, have you solved it? And I set margin of tripletloss =1.0, this still does not works for me.
